schemagic/core.py

```diff
@@ -5,36 +5,48 @@
 
 WHEN_DEBUGGING = lambda: __debug__
 
+
 def validate_map_template(schema, value):
     key_schema, value_schema = schema.items()[0]
-    validate_key_val_pair = lambda key, val: (validate_against_schema(key_schema, key), validate_against_schema(value_schema, val))
+    validate_key_val_pair = lambda key, val: (
+    validate_against_schema(key_schema, key), validate_against_schema(value_schema, val))
     return dict(map(validate_key_val_pair, value.keys(), value.values()))
 
+
 def validate_keyed_mapping(schema, value):
     missing_keys = set(schema.keys()) - set(value.keys())
     if missing_keys:
         raise ValueError("Missing keys {0} for value {1}".format(missing_keys, value))
     return merge_with(validate_against_schema, schema, value)
 
+
 def validate_sequence_template(schema, value):
-    return map(schema[0], value)
+    if hasattr(schema[0], '__call__'):
+        return map(schema[0], value)
+    return map(lambda _: apply(lambda _s, _v: validate_against_schema(_s, _v), _), zip(schema, value))
+
```

Comment:
ah, this is an error, you're correct.  However, we shouldn't ever be directly calling the value, we should always be using "validate_with_schema".  so the if check isnt necessary.  The corrected line should be:

``return map(validate_against_schema, itertools.repeat(schema[0]), value)``